spyder/widgets/ipythonconsole/shell.py

```diff
@@ -341,6 +367,13 @@ def change_mpl_backend(self, command):
             if not 'inline' in command:
                 self.silent_execute(command)
 
+    def capture_dir_change(self, command):
+        """
+        Capture dir change magic for synchronization with working directory.
+        """
+        if command.startswith('%cd') or command.startswith("cd"):
```

Comment:
The second case needs to be improved to not catch false `cd` commands. It can be done with

    ... or command.split()[0] == 'cd'